webknossos-datastore/app/com/scalableminds/webknossos/datastore/dataformats/Cube.scala

```diff
@@ -31,20 +31,23 @@ trait Cube {
     // Check if we are the last one to use this cube, if that is the case and the cube needs to be removed -> remove it
     this.synchronized {
       accessCounter -= 1
-      if (accessCounter == 0 && scheduledForRemoval) {
-        onFinalize()
-      }
+      tryRemoval()
     }
   }
 
   def scheduleForRemoval(): Unit = {
     this.synchronized {
       scheduledForRemoval = true
       // Check if we can directly remove this cube (only possible if it is currently unused)
-      if (accessCounter == 0) {
-        isRemoved = true
-        onFinalize()
-      }
+      tryRemoval()
+    }
+  }
+
+  private def tryRemoval(): Unit = {
+    // should only be called in a synchronized block
+    if (!isRemoved && accessCounter == 0 && scheduledForRemoval) {
```

Comment:
Here we additionally check now that we only try to remove this once. Surely only a cosmetic change, but it could theoretically happen.